include/wx/generic/private/listctrl.h

```diff
@@ -813,6 +814,9 @@ class wxListMainWindow : public wxWindow
     wxColour            *m_highlightColour;
     wxImageList         *m_small_image_list;
     wxImageList         *m_normal_image_list;
+
+    wxVector<wxBitmapBundle> m_normal_images;
```

Comment:
There shouldn't be any need to store these images here, they're already stored in `m_normal_image_list`, why do we need to duplicate them?

Reply:
@vadz I tried to use ```wxWithImages```. My problem is that you have both normal and small image list objects within one class. If I derive ```wxListMainWindow``` from the class ```wxWithImages``` additionally (like seen for other classes) I would only have one instance I could deal with. What I could do is to create a new class derived from ```wxWithImages``` and create two instances from that (not sure if that would be a good design choice). I also saw ```wxWithImages``` is meant to be not copied (no proper copy-constructors). So I am also a bit unsure about the design of that ```wxWithImages``` on that side. 

Can you guide me a bit on how you think it should be implemented properly seen? I'd be happy to get it coded as intended. At least what I have here works with higher DPI, but only on normal icon size. I tried with small icons but ran into a bunch of crashes, so I left those changes out for now... (but maybe it can be implemented properly as well). I tested this against your samples and it looks good and I also tested it against Money Manager EX and it really gives the application a quality push ;-)

Let me know your thoughts :-)

Reply:
We indeed need to do some more refactoring done here, as we must get rid of `m_{small,normal}_image_list` in this class and use `wxListCtrlBase::m_images{Small,Normal}` instead, as we won't be able to deal with high DPI icons correctly as long as `wxImageList` is used directly -- we should only use it via `wxWithImages`. To minimize the number of changes we probably should make the existing `wxImageList*` members `wxWithImages*` instead and make them point to the existing base class objects.

Would this work?

Reply:
Okay, I tried your suggestions and so far it seems to work. Hope the tests pass. Let's see...

Reply:
Looking good. Let me know your thoughts.